Dockerfile

```diff
@@ -1,23 +1,55 @@
-FROM node:12-buster
-
-RUN set -e; \
-    apt update; \
-    apt install -y gettext; \
-    rm -rf /var/lib/apt/lists/*
+FROM node:12-alpine AS BUILD_IMAGE
 
 ARG branch=master
 
 ENV NODE_ENV production
 WORKDIR /opt/magic_mirror
 
-RUN git clone --depth 1 -b ${branch} https://github.com/MichMich/MagicMirror.git .
-RUN cp -R modules /opt/default_modules
-RUN cp -R config /opt/default_config
-RUN npm install --unsafe-perm --silent
+# get magic mirror
+RUN apk update && apk add --no-cache git \
+    && git clone --depth 1 -c advice.detachedHead=false -b ${branch} https://github.com/MichMich/MagicMirror.git . \
+    && apk del git
```

Comment:
we are in the build stage, there is no need to keep this image small by deleting git, we could also use the non alpine base image, which includes git.

Reply:
True. I will remove `apk del git` in the next commit. Nevertheless, I would prefer to stick with the same base image and definitely not switch between alpine and ubuntu or debian. Using the same base image in all the stages makes the build actually faster because docker does not have to pull different base images to start the build.

Build times on my machine
| Stage 0 | Stage 1 | time |
|---------|---------|-----|
| node:12 | node:12-alpine | 225,53 s |
| node:12-alpine | node:12-alpine | 139,66 s |

Between the builds I did run `docker system prune --all`to start from scratch. The time was recorderd with `time docker build -t mm:latest .`